carbon-design-system/carbon
# refactor(upgrade): update CLI deps and setup

This PR updates our old upgrade implementation to bring in newer versions of dependencies and a new project structure. Specifically, we have:

- Project and Workspace for describing a project with workspaces along with an individual project
- Migration for helping to determine what migrations can be applied to a project and its workspaces
- Planner to ask the user what migrations they want to be applied
- Runner to run the migrations and update package.json files

#### Changelog

**New**

**Changed**

- Updated `packages/upgrade` to new approach described above

**Removed**

#### Testing / Reviewing

- Run `./packages/upgrade/bin/carbon-upgrade.js`
- Verify the following output

<details>
<summary>Video of output</summary>

https://user-images.githubusercontent.com/3901764/120839746-ef5dc700-c52e-11eb-802a-1e7995f8257d.mov

</details>
